pytorch_toolkit/ote/ote/metrics/classification/common.py

```diff
@@ -0,0 +1,66 @@
+"""
+ Copyright (c) 2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import os
+import subprocess
+
+from ote import MMACTION_TOOLS
+
+
+def collect_accuracy(path):
+    """ Collects accuracy values in log file. """
+
+    accuracies = []
+    content = '/mean_top1_acc:'
+    with open(path) as input_stream:
+        for line in input_stream:
+            candidate = line.strip()
+            if content in candidate:
+                accuracies.append(float(candidate.split(' ')[-1]))
+
+    return accuracies
+
+
+def mean_accuracy_eval(config_path, work_dir, snapshot, update_config, **kwargs):
+    """ Computes mean accuracy. """
+
+    outputs = []
+
+    try:
+        test_py_stdout = os.path.join(work_dir, 'test_py_stdout')
+
+        update_config = ' '.join([f'{k}={v}' for k, v in update_config.items()])
+        update_config = f' --update_config {update_config}' if update_config else ''
+        update_config = update_config.replace('"', '\\"')
+
+        subprocess.run(
+            f'python {MMACTION_TOOLS}/test.py'
+            f' {config_path} {snapshot}'
+            f' --eval mean_top_k_accuracy'
+            f'{update_config}'
+            f' | tee {test_py_stdout}',
+            check=True, shell=True)
+
+        average_precision = collect_accuracy(os.path.join(work_dir, 'test_py_stdout'))[0]
+        outputs.append({
+            'key': 'accuracy', 'value': 1e2 * average_precision, 'unit': '%', 'display_name': 'Top-1 accuracy'
+        })
+    except:
```

Comment:
What kind of exception is possible?

Reply:
Refactored in favour of `if ... else` construction.